src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -181,3 +183,18 @@ interface TriggeredInfo {
     val user: User?
     val build: Build?
 }
+
+interface FinishBuildTrigger {
+    val initiatedBuildConfiguration: BuildConfigurationId
+    val afterSuccessfulBuildOnly: Boolean
+    val branchFilter: List<BranchRule>
```

Comment:
It names this way in TeamCity UI ([docs](https://confluence.jetbrains.com/display/TCD10/Configuring+VCS+Triggers#ConfiguringVCSTriggers-BranchFilterbranchFilter)).
I agree those interface will be removed, but I thought it will be associated in future with ArtifactRule, VcsTriggerRule (which isn't implemented yet) and other similar rules (i'm not sure there are any others) in abstract class Rule, because they looks similar. But I can't say any valuable  pros of this unification, so now your proposal looks more worthwhile.